### PR TITLE
New 'Log Entry Event Stream' tab & lwc

### LINKED ISSUE #89: Add a lwc to display a Log Entry Event stream
There needs to be a way to easily subscribe to the `LogEntryEvent__e` platform event stream and display it on screen - admins/devs/architects can then use it for real-type monitoring of logs.

The lwc module `empApi` seems like the best option - screen flows don't seem to currently support platform events (flow only supports screenless flows for platform events), so build a custom component instead

https://developer.salesforce.com/docs/component-library/bundle/lightning-emp-api/documentation

### LINKED ISSUE #204: Two Apex tests failing when more than one Apex class named "Logger" is present
I just started evaluating Nebula Logger in our sandbox, and saw a couple Apex test failures I thought you might want to know about.  The class and methods failing are:

Class: LogEntryHandler_Tests
Method: it_should_set_apex_class_details_for_top_level_class (line 299)
Method: it_should_set_apex_class_details_for_inner_class (line 335)

Both lines perform the same query, where "exampleTopLevelClassName" is set to "Logger":

`ApexClass apexClass = [SELECT Id, ApiVersion, CreatedDate, LastModifiedDate FROM ApexClass WHERE Name = :exampleTopLevelClassName];`

These lines fail with "System.QueryException: List has more than 1 row for assignment to SObject" if the org has more than one Apex class named "Logger".  Our org has a managed package installed that also has an Apex class named "Logger", so the query returns two records which cannot be assigned to the single SObject.

It looks like the example "Logger" class it's trying to query is Nebula's own class, so maybe the WHERE clause could be extended to restrict it to one record, such as:

`ApexClass apexClass = [SELECT Id, ApiVersion, CreatedDate, LastModifiedDate FROM ApexClass WHERE Name = :exampleTopLevelClassName AND (NamespacePrefix = NULL OR NamespacePrefix = 'Nebula')];`

That would allow for selecting just Nebula's own Logger class, whether Nebula is installed as an unlocked or managed package.

Anyway, just a thought on a possible fix.

Lastly, I just wanted to give everyone working on this project a huge **THANKS!**  The home-grown logging system I created for our org has hit a limit on its usefulness for troubleshooting more complex logic, and I was faced with either putting a lot of time into enhancing it, or looking for a new logging framework when I ran across Nebula Logger.  The amount of functionality and quality you all have put into Nebula Logger is amazing, and it's greatly appreciated!

### LINKED ISSUE #205: LoggerSettings creation fails as part of a wire function call
When we installed the unlocked version of Nebula, we ran into an error with the following logs:

```14:46:47.0 (9447626)|USER_DEBUG|[45]|INFO|Logger.getTransactionId() == TID:2829900000cbb11d28
14:46:47.0 (10466217)|DML_BEGIN|[2645]|Op:Insert|Type:LoggerSettings__c|Rows:1
14:46:47.0 (10580196)|EXCEPTION_THROWN|[2645]|System.LimitException: Too many DML statements: 1
14:46:47.0 (10880657)|SYSTEM_MODE_EXIT|false
14:46:47.0 (10997450)|SYSTEM_MODE_EXIT|true
14:46:47.0 (11127246)|FATAL_ERROR|System.LimitException: Too many DML statements: 1

Class.Logger.setupOrgDefaults: line 2645, column 1
Class.Logger: line 47, column 1
Class.LWC_CaseHighlights.getCaseDetails: line 18, column 1
14:46:47.0 (11178717)|FATAL_ERROR|System.LimitException: Too many DML statements: 1

Class.Logger.setupOrgDefaults: line 2645, column 1
Class.Logger: line 47, column 1
Class.LWC_CaseHighlights.getCaseDetails: line 18, column 1
14:46:47.11 (11198328)|CUMULATIVE_LIMIT_USAGE
14:46:47.11 (11198328)|LIMIT_USAGE_FOR_NS|(default)|
  Number of SOQL queries: 0 out of 100
  Number of query rows: 0 out of 50000
  Number of SOSL queries: 0 out of 20
  Number of DML statements: 1 out of 0 ******* CLOSE TO LIMIT
  Number of Publish Immediate DML: 0 out of 150
  Number of DML rows: 0 out of 10000
  Maximum CPU time: 0 out of 10000
  Maximum heap size: 0 out of 6000000
  Number of callouts: 0 out of 100
  Number of Email Invocations: 0 out of 10
  Number of future calls: 0 out of 50
  Number of queueable jobs added to the queue: 0 out of 50
  Number of Mobile Apex push calls: 0 out of 10
```

This trace of this error is happening as part of a `@wire` function call:

```
    @AuraEnabled(Cacheable=true)
    public static DTO_CaseHighlightsResponse getCaseDetails( Id caseId, List<String> fieldPaths ) {
        final String METHOD_NAME = 'getCaseDetails()';
        Logger.debug( new LogMessage( LoggerConstants.METHOD_LOG_UNFORMATTED, CLASS_NAME, METHOD_NAME, LoggerConstants.IM_HERE ));

        try {
            // snip
        }
        finally {
            Logger.saveLog();
        }
    }
```

It appears that this is related to certain contexts not being able to do DML (e.g. create the initial LoggerSettings, if none exists)

https://www.levelupsalesforce.com/lwc-too-many-dml-statements-1

Creating the LoggerSettings__c entry (org defaults in our case) immediately stops this failure.

### PR DESCRIPTION
**Pull request by jongpie**
Closes #89 by introducing a new lightning web component `logEntryEventStream` & tab to display `LogEntryEvent__e` records in real-time

- Use custom filters to apply filter criteria to the `LogEntryEvent__e` records displayed
- Buttons within lwc provide additional functionality
  - Toggle Stream button - provides the ability to pause & resume live-streaming of `LogEntryEvent__e` records
  - Clear button - clears any `LogEntryEvent__e` records that are currently displayed on the screen

Other changes
- Fixes #204 - Improved approach for querying for an Apex class in `LogEntryHandler_Tests` to handle orgs that have 2 `Logger` classes (one with namespace and another without namespace)
- Fixes #205 - Eliminated DML on `LoggerSettings__c` within `Logger`, improved in-memory creation of default settings

 ### Example of New 'Log Entry Event Stream' Tab

image

### CODE REVIEW

**Comment by jongpie** [comment]
@jamessimone if you have time to review, I'd love to hear your thoughts on this PR. I'm planning a few future enhancements, but want to wrap up this PR since it's been open for 2+ weeks already. A couple of future enhancements I'm considering: 

- In this PR, I commented-out an "expand/collapse" button that has some issues on different resolutions - I don't feel like fixing right now, so will revisit later
- Eventually, I want to add a way to toggle between the current "console" view and a view that uses a datatable
- The left-side panel (Stream Settings) should be collapsible, but it seemed more complex than I had hoped, so I didn't include it in this PR

Let me know your thoughts!

File: extra-tests/tests/Logger_Tests_ExperienceSite.cls [hunk 1]
```diff
@@ -9,12 +9,23 @@
  */
 @IsTest
 private class Logger_Tests_ExperienceSite {
+    private static final Boolean IS_EXPERIENCE_CLOUD_ENABLED = Schema.getGlobalDescribe().containsKey('Network');
```
Comment (jamessimone) [inline extra-tests/tests/Logger_Tests_ExperienceSite.cls:12]: nice

Comment (jongpie) [reply extra-tests/tests/Logger_Tests_ExperienceSite.cls:12]: Yeah, I don't typically enable experience cloud/communities in my "local" scratch orgs, so these tests were previously always failing, and I got tired of that 😆

File: nebula-logger/main/log-management/lwc/logEntryEventStream/logEntryEventStream.html [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    <lightning-button-group>
+                        <lightning-button
+                            data-id="clear"
+                            label="Clear"
+                            icon-name="utility:delete"
+                            name="clear"
+                            onclick={onClear}
+                            variant="destructive"
+                        ></lightning-button>
+                        <!-- <lightning-button-stateful
```
[the remark this answers is not in the document]
Comment (jongpie) [reply nebula-logger/main/log-management/lwc/logEntryEventStream/logEntryEventStream.html:80]: I think I'm going to leave it in/commented-out for now - the basics are working, but there are a few bugs with it right now, so I was thinking I'd revisit it in a future PR to finish implementing it.

File: nebula-logger/main/log-management/lwc/logEntryEventStream/__tests__/logEntryEventStream.test.js [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+    it('streams a single log entry event', async () => {
+        const element = createElement('log-entry-event-stream', {
+            is: LogEntryEventStream
+        });
+        document.body.appendChild(element);
+        await Promise.resolve();
```
Comment (jamessimone) [inline nebula-logger/main/log-management/lwc/logEntryEventStream/__tests__/logEntryEventStream.test.js:55]: this is a nice pattern!

**Review by jamessimone — APPROVED** [review]
I think this looks good! You answered the only questions I had, and the event stream itself looks great

**Comment by jongpie** [comment]
@jamessimone thanks for the review! I'll merge & release this ASAP

**Pull request merged**